### DISCUSSION TITLE
QUESTION: Preforming Multiple Rollups on Same Object

### DISCUSSION THREAD

**Discussion by AndyHaas** [jamessimone/apex-rollup#183] [Q&A] [answered]
I am looking for how to best use this tool in where I have 4 fields that need to be calculated to the Account object from the Opportunity Record.

LTD Total Revenue
LTD Total Rental
YTD Total Revenue
YTD Total Rental

When I am looking at the Flow action “Perform Rollup__mdt-based Rollup” how does that know which rollup to run or will it run all of the CMDT configurations I have configured?

**Comment by jamessimone** [comment]
It will run for all CMDT configurations you have configured, yes. The logic
works (simply) like this - any CMDT whose Calc Item is set to the current
flow/Apex trigger calling Rollup OR the same thing for any CMDT whose
Lookup Object matches and the CMDT record's "Is Rollup Started From Parent"
field is flagged

**Comment by AndyHaas** [comment]
Ah, so this has to be used within a Flow Trigger? Could I set this up under an auto-launch flow and call it from a Flow Trigger as a subflow?

Or is it just the Object for "Prior records to rollup" (Input) that signifies which CMDT to run?

**Comment by jamessimone** [comment, marked as answer]
@AndyHaas I don't know that you'll be able to set this up within an auto-launched flow since the type for each object is required to be selected prior to being able to pass in each of the collection variables themselves, but I haven't actually tried this since Winter22 came out and it may be possible.

Otherwise, you have two options:

- use a RT flow, as you said, without the use of a subflow (downside being you'd have to add the rollup action to each RT flow you have)
- use the Apex one-liner (if you have Apex triggers/trigger handlers already, there really isn't much of a downside to this approach)

But yes - either way - Rollup looks at the records being passed to it in order to determine which CMDT records to run.
